src/platform/tls_openssl.c

```diff
@@ -0,0 +1,3457 @@
+    //
+    CXPLAT_SEC_CONFIG* SecConfig;
+
+    //
+    // Pointer to HKDF label definitions used in the key derivation process.
+    //
+    const QUIC_HKDF_LABELS* HkdfLabels;
+
+    //
+    // Indicates if the endpoint is acting as a server.
+    //
+    BOOLEAN IsServer : 1;
+
+    //
+    // Indicates if a peer certificate has been received.
+    //
+    BOOLEAN PeerCertReceived : 1;
+
+    //
+    // Indicates if the peer's transport parameters have been received.
+    //
+    BOOLEAN PeerTPReceived : 1;
+
+    //
+    // QUIC transport parameter extension type used in the session.
+    //
+    uint16_t QuicTpExtType;
+
+    //
+    // Length of the ALPN buffer.
+    //
+    uint16_t AlpnBufferLength;
+
+    //
+    // Pointer to the ALPN buffer data.
+    //
+    const uint8_t* AlpnBuffer;
+
+    //
+    // Pointer to the Server Name Indication (SNI) string.
+    //
+    const char* SNI;
+
+    //
+    // OpenSSL SSL object used for the TLS handshake and encryption.
+    //
+    SSL *Ssl;
+
+    //
+    // Pointer to internal TLS processing state.
+    //
+    CXPLAT_TLS_PROCESS_STATE* State;
+
+    //
+    // Flags indicating the results of TLS processing.
+    //
+    CXPLAT_TLS_RESULT_FLAGS ResultFlags;
+
+    //
+    // Pointer to the QUIC connection associated with this TLS session.
+    //
+    QUIC_CONNECTION* Connection;
+
+    //
+    // Pointer to derived TLS secrets for encryption and decryption.
+    //
+    QUIC_TLS_SECRETS* TlsSecrets;
+
+} CXPLAT_TLS;
+
+//
+// @struct RECORD_ENTRY
+// @brief Represents a buffered SSL record in a linked list.
+//
+// This structure is used to store an SSL record along with its
+// metAData and linkage in a list. It supports tracking incomplete
+// records and whether the memory should be freed.
+//
+typedef struct RECORD_ENTRY {
+    //
+    // Linked list node for linking Entries in a list.
+    //
+    CXPLAT_LIST_ENTRY Entries;
+
+    //
+    // Pointer to the raw SSL record data.
+    //
+    unsigned char *Record;
+
+    //
+    // Length of the SSL record.
+    //
+    size_t RecLen;
+
+    //
+    // Pointer to the associated SSL connection.
+    //
+    SSL *Ssl;
+
+    //
+    // Non-zero if the record is incomplete.
+    //
+    unsigned char Incomplete;
+
+    //
+    // Non-zero if the record memory should be freed.
+    //
+    unsigned char FreeMe;
+} RECORD_ENTRY;
+
+//
+// @struct AUX_DATA
+// @brief holds auxilliary data we need for each ssl
+//
+typedef struct AUX_DATA {
+    //
+    // @brief transport params for our endpoint 
+    //
+    const uint8_t *Tp;
+
+    //
+    // @brief peer transport params
+    //
+    uint8_t *PeerTp;
+
+    //
+    // @brief peer transport param len
+    //
+    size_t PeerTpLen;
+
+    //
+    // @brief The current encryption level we are sending data for
+    //
+    uint32_t Level;
+
+    //
+    // @brief this SSL's receive record list
+    //
+    CXPLAT_LIST_ENTRY RList;
```

Comment:
Let's make this variable name a bit more descriptive please. 😄 `RecordList` for instance.